backend/api/battles/serializers.py

```diff
@@ -57,7 +79,57 @@ def validate(self, attrs):
         if pokemon_points_sum > 600:
             raise serializers.ValidationError("Pokemons' points sum cannot be more than 600")
 
+        if self.instance:
+            object_id = self.instance.id
+
+            battle = Battle.objects.get(pk=object_id)
+            if request.user != battle.opponent:
+                raise serializers.ValidationError(
+                    "You can't update opponent Pokemons if you aren't the battle opponent"
+                )
+
+            opponent_pokemon_1 = attrs.get("opponent_pokemon_1")
+            opponent_pokemon_2 = attrs.get("opponent_pokemon_2")
+            opponent_pokemon_3 = attrs.get("opponent_pokemon_3")
+
+            pokemon_points_sum = 0
+            try:
+                pokemon_points_sum = get_pokemons_points_sum(
+                    [opponent_pokemon_1.name, opponent_pokemon_2.name, opponent_pokemon_3.name]
+                )
+            except PokemonNotFound:
+                pass
+
+            if pokemon_points_sum > 600:
+                raise serializers.ValidationError("Pokemons' points sum cannot be more than 600")
+
         return attrs
 
     def create(self, validated_data):
         return Battle.objects.create(**validated_data)
+
+    def update_opponent_pokemons(
```

Comment:
Why do you need this? I though the update would work out of the box. Am I missing something?

Reply:
I thought it could be helpful if, in the future, I want to update different battle data. But I understand your point. Probably it was an overengineering as Pokebattle isn't in this scenario.